File: ToDoList/ToDoList/Task.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ToDoList
{
    enum Priority
    {
        Else,       // 0
        High,       // 1
        Medium,     // 2
        Low,        // 3
    }
    class Task
    {
        //fields
        private static int currentId = 1;

        //properties
        public int TaskId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime DueDate { get; set; }
        public Priority Priority { get; set; }

        //constructors
        public Task(string name, string description, DateTime startDate, DateTime dueDate, Priority priority)
        {
            TaskId = currentId;
            currentId++;
            Name = name;
            Description = description;
            StartDate = startDate;
            DueDate = dueDate;
            Priority = priority;
        }
        public Task(string name, string description, Priority priority)
        {
            TaskId = currentId;
            currentId++;
            Name = name;
            Description = description;
            StartDate = DateTime.Today;
            DueDate = DateTime.Today;
            Priority = priority;
        }
        //methods
        public override string ToString()
        {
            return $"currentId : {currentId}\nTaskId: {TaskId}\nName: {Name}\nDescription: {Description}\nStartDate: {StartDate}\nDueDate: {DueDate}\nPriority: {Priority}";
        }
    }
}

File: Java/ToDoList/src/Task.java
import java.time.LocalDate;
import java.time.format.DateTimeFormatter;

enum Priority
{
	LOW,
	MEDIUM,
	HIGH
}

public class Task //(need to be changed to abstract!!!!!!)
{	
	//Fields
	static int taskId = 1;
	protected int id;
	protected String name;
	protected String description;
	protected Priority priority;
	protected LocalDate dueDate = null;
	protected LocalDate created; 
	
	protected DateTimeFormatter dateFormat = DateTimeFormatter.ofPattern("dd-MM-yyyy");
	
	//constructors (with DueDate, withoutDueDate)
	//without DueDate
	public Task(String name,String description, Priority priority)
	{		
		id = taskId++;
		this.name = name;
		this.description = description;
		this.priority = priority;
		created = LocalDate.now();;
	}
	//with DueDate
	public Task(String name,String description, Priority priority, String dueDateString)
	{		
		id = taskId++;
		this.name = name;
		this.description = description;
		this.priority = priority;
		created = LocalDate.now();;
		setDueDate(dueDateString);
	}
	
	//Getter and Setter
	//name
	public String getName()
	{
		return name;
	}
	public void setName(String name)
	{
		this.name = name;
	}
	//description
	public String getDescription()
	{
		return description;
	}
	public void setDescription(String description)
	{
		this.description = description;
	}
	//priority
	public Priority getPriority()
	{
		return priority;
	}
	public void setPriority(Priority priority)
	{
		this.priority = priority;
	}
	//created: read-only
	public String getCreated()
	{
		return created.format(dateFormat);
	}
	//dueDate: get String type and formatting as LocalDate type.
	public LocalDate getDueDate()
	{
		return dueDate;
	}
	public void setDueDate(String userInput)
	{
		if(userInput == "No" || userInput == "NO" || userInput == "no"){
			this.dueDate = null;
		}
		else{
			DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy");
			LocalDate dueDate = LocalDate.parse(userInput, formatter);
			
			this.dueDate = dueDate;
		}
	}
	
	@Override
	public String toString()
	{
		//String s = String.format("$%.2f", salary);
		String s = String.format("#%d - %s - %s", id, priority, name);
		return s;
	}
}
